R/foverlaps.R

```diff
@@ -29,19 +29,17 @@ foverlaps <- function(x, y, by.x = if (!is.null(key(x))) key(x) else key(y), by.
   }
   if (is.numeric(by.y)) {
     if (any(by.y < 0L) || any(by.y > length(y)))
-      stop("Invalid numeric value for 'by.x'; it should be a vector with values 1 <= by.y <= length(y)")
+      stop("Invalid numeric value for 'by.y'; it should be a vector with values 1 <= by.y <= length(y)")
     by.y = names(y)[by.y]
   }
-  if (!length(by.x) || !is.character(by.x))
-    stop("A non-empty vector of column names is required for by.x")
-  if (!length(by.y) || !is.character(by.y))
-    stop("A non-empty vector of column names is required for by.y")
+  if (!is.character(by.x))
+    stop("A non-empty vector of column names or numbers is required for by.x")
+  if (!is.character(by.y))
+    stop("A non-empty vector of column names or numbers is required for by.y")
   if (!identical(by.y, key(y)[seq_along(by.y)]))
-    stop("The first ", length(by.y), " columns of y's key is not identical to the columns specified in by.y.")
+    stop("The first ", length(by.y), " columns of y's key must be identical to the columns specified in by.y.")
   if (anyNA(chmatch(by.x, names(x))))
     stop("Elements listed in 'by.x' must be valid names in data.table 'x'")
-  if (anyNA(chmatch(by.y, names(y))))
```

Comment:
this is not possible after checking `(!identical(by.y, key(y)[seq_along(by.y)]))`